### PR TITLE
saltpack bug fixes

### PR DESCRIPTION
**Pull request by AMarcedone**

### CODE REVIEW

**Comment by AMarcedone** [comment]
This covers 
https://keybase.atlassian.net/browse/CORE-2305
https://keybase.atlassian.net/browse/CORE-3404
https://keybase.atlassian.net/browse/CORE-3455
https://keybase.atlassian.net/browse/CORE-4112 
https://keybase.atlassian.net/browse/CORE-8496 
https://keybase.atlassian.net/browse/CORE-8498

File: go/client/saltpack_ui.go [hunk 1]
```diff
@@ -95,23 +113,22 @@ func (s *SaltpackUI) SaltpackVerifySuccess(_ context.Context, arg keybase1.Saltp
 	var un string
 	switch arg.Sender.SenderType {
 	case keybase1.SaltpackSenderType_UNKNOWN:
-		un = "The signer of this message is unknown to Keybase"
-	case keybase1.SaltpackSenderType_TRACKING_OK, keybase1.SaltpackSenderType_NOT_TRACKED:
+		un = fmt.Sprintf("The signer of this message is unknown to Keybase.\nSigning key ID: %s", arg.SigningKID)
+	case keybase1.SaltpackSenderType_TRACKING_OK:
 		un = fmt.Sprintf("Signed by %s", ColorString(s.G(), "bold", arg.Sender.Username))
+	case keybase1.SaltpackSenderType_NOT_TRACKED:
+		un = fmt.Sprintf("Signed by %s (which you do not follow)", ColorString(s.G(), "bold", arg.Sender.Username))
```
Comment (maxtaco) [inline go/client/saltpack_ui.go]: `which` -> `whom`

File: go/client/ui.go [hunk 2]
```diff
@@ -507,17 +515,18 @@ func (ui BaseIdentifyUI) FinishWebProofCheck(p keybase1.RemoteProof, l keybase1.
...
-
+	ui.displayedProofs = true
 	return nil
 }
 
-func (ui BaseIdentifyUI) DisplayCryptocurrency(l keybase1.Cryptocurrency) error {
+func (ui *BaseIdentifyUI) DisplayCryptocurrency(l keybase1.Cryptocurrency) error {
 	msg := (BTC + " " + " " + l.Family + " " + ColorString(ui.G(), "green", l.Address))
 	ui.ReportHook(msg)
+	ui.displayedProofs = true
```
Comment (maxtaco) [inline go/client/ui.go]: this doesn't prove anything, i wouldn't include this.

**Comment by AMarcedone** [comment]
@maxtaco fixed! Waiting for CI / your final approval.

**Pull request merged**